services/rest/src/main/java/mil/nga/giat/geowave/service/rest/GeoWaveOperationServiceWrapper.java

```diff
@@ -189,63 +189,71 @@ private void injectParameters(
 
 			Object objValue = null;
 			Class<?> type = f.getType();
-
-			if (List.class.isAssignableFrom(type)) {
-				objValue = requestParameters.getList(f.getName());
-			}
-			else if (type.isArray()) {
-				objValue = requestParameters.getArray(f.getName());
-				if (objValue != null) {
-					objValue = Arrays.copyOf(
-							(Object[]) objValue,
-							((Object[]) objValue).length,
-							f.getType());
-				}
-			}
-			else {
-				final String strValue = (String) requestParameters.getString(f.getName());
-				if (strValue != null) {
-					if (Long.class.isAssignableFrom(type) || long.class.isAssignableFrom(type)) {
-						objValue = Long.valueOf(strValue);
-					}
-					else if (Integer.class.isAssignableFrom(type) || int.class.isAssignableFrom(type)) {
-						objValue = Integer.valueOf(strValue);
-					}
-					else if (Short.class.isAssignableFrom(type) || short.class.isAssignableFrom(type)) {
-						objValue = Short.valueOf(strValue);
-					}
-					else if (Byte.class.isAssignableFrom(type) || byte.class.isAssignableFrom(type)) {
-						objValue = Byte.valueOf(strValue);
-					}
-					else if (Double.class.isAssignableFrom(type) || double.class.isAssignableFrom(type)) {
-						objValue = Double.valueOf(strValue);
-					}
-					else if (Float.class.isAssignableFrom(type) || float.class.isAssignableFrom(type)) {
-						objValue = Float.valueOf(strValue);
-					}
-					else if (Boolean.class.isAssignableFrom(type) || boolean.class.isAssignableFrom(type)) {
-						objValue = Boolean.valueOf(strValue);
+			Field field = f.getField();
+			final String strValue = (String) requestParameters.getString(f.getName());
+			
+			if (field.isAnnotationPresent(Parameter.class)) {
+				Class<? extends IStringConverter<?>> converter = field.getAnnotation(
```

Comment:
So right here, the code gets the Class object and tries to see if it is one of the converters, or if it's "NoConverter." It does this for every single field, even though only a few have converters, which seems inefficient to me. However, it is important to do this before some of the other checks, because there are certain cases where the other checks would catch the type. I.E. passwordCoverter converts a string to another string, so that would get caught in the String check

Reply:
seems reasonable enough, I'd check for converter == null at the front of the if statement even though apparently it must always get set to "NoConverter" in practice just to add a little more protection from NPEs